### ISSUE TITLE
How to use a proxy endpoint

### ISSUE THREAD

**Issue by erfan-azhdari** [J-Rios/uTLGBotLib-arduino#5]
Telegram is unfortunately censored in Iran.
Is there any way that I can use a proxy server instead of `api.telegram.org`?

**Comment by J-Rios** [comment]
Hi, current version of the  library doesn't support the use of a Proxy, it should be modified to use it.

**Comment by erfan-azhdari** [comment]
How can I modify the library?
Is editing [this](https://github.com/J-Rios/uTLGBotLib-arduino/blob/5d4e1e2d9787a28f89a5a5805f98703879f934ab/src/utlgbotlib.h#L66)(`#define TELEGRAM_HOST "api.telegram.org"`) enough?
what about [this](https://github.com/J-Rios/uTLGBotLib-arduino/blob/master/res/certs/apitelegramorg.crt)(`apitelegramorg.crt`) certificate?
My proxy server is hosted on Heroku and I'm trying to run this on `ESP32` or `ESP8266`.

**Comment by erfan-azhdari** [comment]
I modified the library and changed the public key and the fingerprint but it still doesn't work it fails with

`[Bot] Trying to send getUpdates request...`
`[HTTPS] POST request successfully sent.`
`[HTTPS] Waiting for response...`
`[HTTPS] Response successfully received.`
`[Bot] Response received:`
`{"update_id":950784443,"message":{"message_id":28,"from":{"id":123456,"is_bot":false,"first_name":"____","language_code":"en"},"chat":{"id":123456,"first_name":"____","type":"private"},"date":1594999322,"text":"1"}}]}`
`Can't parse JSON data. Code -2`
`[Bot] Error: Bad JSON sintax from received response.`

**Comment by erfan-azhdari** [comment]
The JSON seems to be invalid but it's not what the API returns.
both my mirror and the telegram API return the same output.

**Comment by J-Rios** [comment]
Are you using released version?
Have you modified something else of the library code?

That error tell us that the telegram server connecting ok, so only json update parse fail...

**Comment by erfan-azhdari** [comment]
I am using the version on arduino's servers. 
I changed `TELEGRAM_HOST` and it's length.
I also changed the .CRT certificate and the header file header file in src/ for the certificate root key and signature.
Although it didn't make sense because the library doesn't use SSL pinning in arduino at all.

**Comment by J-Rios** [comment]
That's correct, released version for Arduino usage doesn't uses SSL/TLS client/server certificates identity trust verification.

**Comment by erfan-azhdari** [comment]
What should I do now? 
Should I use this repo's master branch?

**Comment by J-Rios** [comment]
You can try it (however your problem is not related to certs), but master branch of [full uTLGBot library](https://github.com/J-Rios/uTLGBotLib). It should support server certificate verification, however note that this is not realeased and is not complete validated yet, so could have bugs on it.

**Comment by erfan-azhdari** [comment]
I don't care about SSL pinning.
It can be done by uncommenting two lines in the http library.
I want to resolve the JSON issue(invalid syntax).
